osgar/drivers/logsocket.py

```diff
@@ -202,6 +213,31 @@ def run_input(self):
             if self.sleep is not None:
                 self.bus.sleep(self.sleep)
 
+    def run_output(self):
+        try:
+            while True:
+                __, channel, out_data = self.bus.listen()
+                if channel == 'raw':
+                    if isinstance(out_data, list):
+                        assert len(out_data) == 3, out_data  # unsupported output data
+                        url, header, data = out_data
+                        request = urllib.request.Request(url, data=data, headers=header, method='POST')
+                    elif isinstance(out_data, str):  # url only
+                        request = out_data
+                    else:
+                        assert False, out_data  # unsupported output data
+
+                    with urllib.request.urlopen(request, timeout=0.5) as f:
```

Comment:
the timeout should be part of module config, and 0.5s maybe as default? something like `request_timeout_sec`??